problem6.js

```diff
@@ -0,0 +1,99 @@
+let rxjs = require("rxjs") ; 
+let events = require("events");
+
+function between(min, max) {  
+    return Math.floor(
+      Math.random() * (max - min) + min
+    )
+  }
+
+function emitValue(event,range){
+    setInterval(() => {
+        event.emit("data",between(range[0],range[1]));
+    }, between(100,2000));
+}
+
+const getTemperature = ()=> {
+    let temperatureEmitter = new events.EventEmitter();    
+    emitValue(temperatureEmitter,[0,10]);
+    return temperatureEmitter ;
+}
+
+const  getAirPressure= ()=>{
+    let airPressureEmitter = new events.EventEmitter();    
+    emitValue(airPressureEmitter,[10,20]);
+    return airPressureEmitter ;
+}
+
+const getHumidity= ()=>{
+    let humidityEmitter = new events.EventEmitter();    
+    emitValue(humidityEmitter,[20,30]);
+    return humidityEmitter ;
+}
+
+
+const setValue = (obj,prop) => {
+  return  setTimeout(() => {
+      obj[prop] = "N/A" ;
+  }, 1000);
+}
+
+const observable$ =new rxjs.Observable(subscriber =>{
+    let obj ={
+        temperature : null ,
+        airPresure : null,
+        humidity : null
+    };
+
+    getTemperature().on("data",value=>{ 
+            obj.temperature = value
+            if (obj.temperature != null && obj.airPresure != null && obj.humidity != null){
```

Comment:
Rather than comparing to != null, !obj.temperature would check for both undefined and null.